src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs

```diff
@@ -318,6 +323,8 @@ private JsonConverter GetConverterFromType(Type typeToConvert)
             return converter;
         }
 
+        [UnconditionalSuppressMessage("AotAnalysis", "IL3050:RequiresDynamicCode", Justification = "The factory constructors are only invoked in the context of reflection serialization code paths " +
+            "and are marked RequiresDynamicCode")]
```

Comment:
Why isn't this marked with `RequiresDynamicCode`?

Reply:
This API is a gateway to a much larger dependency graph. I need to understand the potential impact on Json source generated paths better to correctly annotate this and need help from @eiriktsarpalis @layomia @krwq first

Reply:
That is somewhat concerning. If this means the source generation path might not work in NativeAot, we will need to understand it.

Reply:
Should this suppression be replaced by proper annotations instead? 

For example, I see the following callchain:

`[RequiresDynamicCode] public GetConverter`
-> 
`GetConverterInternal` (not annotated)
->
`GetConverterFromType` (not annotated)
->
`[UnconditionalSuppressMessage] GetConverterFromAttribute`

Should `GetConverterInternal` and `GetConverterFromType` be annotated as `RequiresDynamicCode` on this call-chain? (And similar to other call-chains.)

Reply:
There are 2 concerns in the call graph that I left open to get FB;

- [SourceGenJsonTypeInfo<T>](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/SourceGenJsonTypeInfoOfT.cs#L15) has 2 nodes in the path; [ObjectDefaultConverter<T>](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/SourceGenJsonTypeInfoOfT.cs#L84) and an [assert](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/SourceGenJsonTypeInfoOfT.cs#L173). **Is it safe to suppress these calls?**
- [JsonMetadataServices](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonMetadataServices.Converters.cs#L9): Has a call to [ObjectCoverter](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonMetadataServices.Converters.cs#L141) that is used by [GetDefaultSimpleConverters](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs#L95). **Not sure what to do here.**

Reply:
Opened #68878 to track the status since there needs to be converter changes that should be done by domain experts in this space.